src/mono/wasm/runtime/cjs/dotnet.cjs.lib.js

```diff
@@ -4,22 +4,34 @@
 
 "use strict";
 
+#if USE_PTHREADS
+const usePThreads = `true`;
```

Comment:
do you really want string and not bool ?

Reply:
Which tool does the interpolation here? If that was JS it would be \`${true}\`

Reply:
It goes through Emscripten's preprocessor (which takes care of the `#if`), and then `$DOTNET__postset` this whole file runs during emscripten link time. So the `usePThreads` string is interpolated into the `$DOTNET__postset` string (so we get `$DOTNET__postset = "let __dotnet_replacement_PThread_loadWasmModuleToWorker = ${usePThreads} ? PThread.loadWasmModuleToWorker : null..."`.  Then the `$DOTNET__postset` string gets used by the emscripten library manager and gets pasted into the body of the emscripten IIFE.

The upshot is that if we're in `#if USE_PTHREADS`, the body of the emscripten module will contain `let __dotnet_replacement_PThread_loadWasmModuleToWorker = false ? PThread.loadWasmModuleToWorker : undefined;`.  And then rollup collapses that to just `let __dotnet_replacement_PThread_loadWasmModuleToWorker  = undefined` which is what we want when there's no threads.

Reply:
This file is not processed by rollup at all. Maybe closure compiler in emcc does that ?
I understand that the preprocesor would handle `#if`. I don't follow that it would take the code between `#if` and `#else` and evaluate it. I think it would just keep is as is as text, no ?

Reply:
I get it now, the evaluation of `DOTNET__postset` would do the `${isPThread}` evaluation just before it enters closure compiler.